service/matching/task_queue_partition_manager.go

```diff
@@ -358,6 +358,14 @@ func (pm *taskQueuePartitionManagerImpl) unloadPhysicalQueue(unloadedDbq physica
 		// this is the default queue, unload the whole partition if it is not healthy
 		if pm.defaultQueue == unloadedDbq {
 			pm.unloadFromEngine()
+
+			// have to iterate over versionedQueues and call updatePhysicalTaskQueueGauge
+			pm.versionedQueuesLock.Lock()
+			for _, vq := range pm.versionedQueues {
+				vqImpl := vq.(*physicalTaskQueueManagerImpl)
+				pm.engine.updatePhysicalTaskQueueGauge(vqImpl, -1)
```

Comment:
Not sure the existing behavior is correct, we should make sure Stop is called on all versionedQueues of this partition when unloading whole partition due to default queue being unloaded.
If Stop is called on verisoned queues, then the metric should be updated (based on my other comment about moving the updatePhysicalTaskQueueGauge call to the Stop method of physicalTaskQueueManagerImpl)

Reply:
Makes sense. Making the changes.